server/src/main/java/org/candlepin/util/EntityVersioningRetryWrapper.java

```diff
@@ -0,0 +1,109 @@
+/**
+ * Copyright (c) 2009 - 2021 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.util;
```

Comment:
Suggestion: this class could go in `org.candlepin.controller.util`, since it's not general utility, but relevant to the classes in the controller package (Product/ContentManagers, RefreshWorker). (Also, I know it's also being used by the *Resource classes, but I feel like it shouldn't; see my other comment on the resource class)

Reply:
Sure. Once we decide on how to address the other issue, we can figure out where this should go.

Reply:
This has been moved to the suggested package for now